tests/unit/nupic/algorithms/sdr_classifier_test.py

```diff
@@ -0,0 +1,711 @@
+    for _ in xrange(10000):
+      randomNumber = random.random()
+      if randomNumber < 0.3:
+        bucketIdx = 0
+      elif randomNumber < 0.6:
+        bucketIdx = 1
+      else:
+        bucketIdx = 2
+      c.compute(recordNum=recordNum, patternNZ=SDR1,
+                classification={'bucketIdx': bucketIdx, 'actValue': bucketIdx},
+                learn=True, infer=True)
+      recordNum += 1
+
+      randomNumber = random.random()
+      if randomNumber < 0.5:
+        bucketIdx = 1
+      else:
+        bucketIdx = 3
+      c.compute(recordNum=recordNum, patternNZ=SDR2,
+                classification={'bucketIdx': bucketIdx, 'actValue': bucketIdx},
+                learn=True, infer=True)
+      recordNum += 1
+
+    result = c.compute(
+        recordNum=2, patternNZ=SDR1,
+        classification={'bucketIdx': 0, 'actValue': 0},
+        learn=False, infer=True)
+    self.assertAlmostEqual(result[0][0], 0.3, places=1)
+    self.assertAlmostEqual(result[0][1], 0.3, places=1)
+    self.assertAlmostEqual(result[0][2], 0.4, places=1)
+
+    result = c.compute(
+        recordNum=2, patternNZ=SDR2,
+        classification={'bucketIdx': 0, 'actValue': 0},
+        learn=False, infer=True)
+    self.assertAlmostEqual(result[0][1], 0.5, places=1)
+    self.assertAlmostEqual(result[0][3], 0.5, places=1)
+
+
+  def testPredictionDistributionContinuousLearning(self):
+    """ Test the distribution of predictions with overlapping input SDRs
+
+    First, we intend the classifier to learn the associations:
+      SDR1    => bucketIdx 0 (30%)
+              => bucketIdx 1 (30%)
+              => bucketIdx 2 (40%)
+
+      SDR2    => bucketIdx 1 (50%)
+              => bucketIdx 3 (50%)
+
+    After 20000 iterations, we change the association to
+      SDR1    => bucketIdx 0 (30%)
+              => bucketIdx 1 (20%)
+              => bucketIdx 2 (20%)
+              => bucketIdx 4 (30%)
+
+      SDR2    => bucketIdx 1 (50%)
+              => bucketIdx 3 (50%)
+
+    The classifier should adapt continuously and learn new associations
+    """
+
+    c = self._classifier([0], 0.0002, 0.1, 0)
+    recordNum = 0
+
+    SDR1 = [1, 3, 5]
+    SDR2 = [2, 4, 6]
+
+    random.seed(42)
+    for _ in xrange(10000):
+      randomNumber = random.random()
+      if randomNumber < 0.3:
+        bucketIdx = 0
+      elif randomNumber < 0.6:
+        bucketIdx = 1
+      else:
+        bucketIdx = 2
+      c.compute(recordNum=recordNum, patternNZ=SDR1,
+                classification={'bucketIdx': bucketIdx, 'actValue': bucketIdx},
+                learn=True, infer=True)
+      recordNum += 1
+
+      randomNumber = random.random()
+      if randomNumber < 0.5:
+        bucketIdx = 1
+      else:
+        bucketIdx = 3
+      c.compute(recordNum=recordNum, patternNZ=SDR2,
+                classification={'bucketIdx': bucketIdx, 'actValue': bucketIdx},
+                learn=True, infer=True)
+      recordNum += 1
+
+    for _ in xrange(30000):
```

Comment:
Do we really need 10,000 and 30,000 iterations for this test? It would be good if we could do this much quicker.